plugins/in_kafka/in_kafka.c

```diff
@@ -234,6 +269,15 @@ static int in_kafka_init(struct flb_input_instance *ins,
         goto init_error;
     }
 
+    ctx->log_encoder = flb_log_event_encoder_create(FLB_LOG_EVENT_FORMAT_DEFAULT);
+
+    if (ctx->log_encoder == NULL) {
+        flb_plg_error(ins, "could not initialize log encoder");
+        flb_log_event_encoder_destroy(ctx->log_encoder);
+
+        return -1;
+    }
+
     return 0;
 
 init_error:
```

Comment:
It seems that since now we can get to this label after initializing the collector we need to detect that and dispose of it accordingly.

It seems that this plugin doesn't do it in the exit callback either which could cause problems with hot reload so it needs to be fixed.

You can find an example of how to do it in other plugins.